x-pack/plugins/observability_solution/observability_ai_assistant/server/service/util/get_access_query.ts

```diff
@@ -20,7 +20,17 @@ export function getAccessQuery({
               should: [
                 { term: { public: true } },
                 ...(user
-                  ? [{ term: user.id ? { 'user.id': user.id } : { 'user.name': user.name } }]
+                  ? user.id
+                    ? [
+                        { term: { 'user.id': user.id } },
+                        {
+                          bool: {
+                            must_not: { exists: { field: 'user.id' } },
+                            must: { term: { 'user.name': user.name } },
+                          },
+                        },
+                      ]
+                    : [{ term: { 'user.name': user.name } }]
```

Comment:
Perhaps the flattened ternaries are easier to read?

```suggestion
              ...(user?.id
                ? [
                    { term: { "user.id": user.id } },
                    {
                      bool: {
                        must_not: { exists: { field: "user.id" } },
                        must: { term: { "user.name": user.name } },
                      },
                    },
                  ]
                : []),
              ...(user?.name && !user.id ? [{ term: { "user.name": user.name } }] : []),
```

Reply:
Or alternatively a separate function

```ts
function getUserAccessFilters(user?: User) {
  if (!user) {
    return [];
  }

  if (user.id) {
    return [
      { term: { 'user.id': user.id } },
      {
        bool: {
          must_not: { exists: { field: 'user.id' } },
          must: { term: { 'user.name': user.name } },
        },
      },
    ];
  }

  return [{ term: { 'user.name': user.name } }];
}
```

Reply:
a separate function is even easier to read. I prefer this alternative.